src/app/services/backend.service.ts

```diff
@@ -11,6 +11,26 @@ export class BackendService {
     private session: SessionService
   ) { }
 
+  contacts() {
+    return this.http.get('api/contacts').toPromise();
+  }
+
+  addContact(card) {
+    return this.http.post('api/contacts', card).toPromise();
+  }
+
+  contactId(id, contact) {
+    return this.http.get(`api/contacts/${id}`).toPromise();
```

Comment:
the contact parameter is not being used anywhere. Also typically you would just name this method `getContact`